lib/Runtime/PlatformAgnostic/Platform/Linux/PerfTrace.cpp

```diff
@@ -0,0 +1,150 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "Common.h"
+#include "Runtime.h"
+#include "ChakraPlatform.h"
+
+#include <signal.h>
+#include <errno.h>
+#include <unistd.h>
+
+using namespace Js;
+
+//
+//  Handle the SIGUSR2 signal by setting a flag
+//  indicating that we should call WritePerfMap later on
+//
+void handle_signal(int signum)
+{
+    PlatformAgnostic::PerfTrace::mapsRequested = true;
+}
+
+namespace PlatformAgnostic
+{
+
+volatile bool PerfTrace::mapsRequested = false;
+  
+//
+// Registers a signal handler for SIGUSR2 
+//
+void PerfTrace::Register()
+{
+    struct sigaction newAction = {0};
+    newAction.sa_flags = SA_RESTART;
+    newAction.sa_handler = handle_signal;
+    sigemptyset(&newAction.sa_mask);
+
+    struct sigaction previousAction = {0};
+
+    if (-1 == sigaction(PERFMAP_SIGNAL, &newAction, &previousAction))
+    {
+        AssertMsg(errno, "PerfTrace::Register: sigaction() call failed\n");
+    }
+}
+
+void  PerfTrace::WritePerfMap()
+{
+#if ENABLE_NATIVE_CODEGEN
+    // Lock threadContext list during etw rundown
+    AutoCriticalSection autoThreadContextCs(ThreadContext::GetCriticalSection());
+
+    ThreadContext * threadContext = ThreadContext::GetThreadContextList();
+
+    FILE * perfMapFile;
+
+    {
+        const size_t PERFMAP_FILENAME_MAX_LENGTH = 30;
+        char perfMapFilename[PERFMAP_FILENAME_MAX_LENGTH];
+        pid_t processId = getpid();
+        snprintf(perfMapFilename, PERFMAP_FILENAME_MAX_LENGTH, "/tmp/perf-%d.map", processId);
```

Comment:
I've gone with `/tmp/perf-<pid>.map` because that is the format that `perf` expects. If we gave it a different name, you would need to rename it before it worked with other tools.